### PR TITLE
empty collections should not appear in search results/browse all

### LINKED ISSUE #1226: Empty Collections should not appear in search results/browse all
# Feature Request
ScholarSphere users often create collections before they create the works that will (eventually) go in them. As a result, empty collections are often floating around on the "browse all" page.

WDLL: Collections without any published works should not appear in search results. They are similar to draft works in this respect.

### PR DESCRIPTION
**Pull request by banukutlu**
Closes #1226

### CODE REVIEW

File: app/models/collection.rb [hunk 1]
```diff
@@ -160,6 +160,12 @@ def thumbnail_present?
     thumbnail_urls.present?
   end
 
+  def draft?
```
Comment (banukutlu) [inline app/models/collection.rb]: @nicktiberi @rschenk pls let me know if you can think of a better name for this method, it is basically to check if the collection has any published work or not.

File: app/schemas/collection_schema.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class CollectionSchema < BaseSchema
```
Comment (banukutlu) [inline app/schemas/collection_schema.rb:3]: @nicktiberi @rschenk same here, better schema and field name suggestion are more than welcome :) 
also open to suggestions re if this field could be part another schema or not?

File: app/models/collection.rb [hunk 3]
```diff
@@ -160,6 +160,12 @@ def thumbnail_present?
...
+    works
+      .select { |work| work.representative_version.aasm_state == 'published' }
```
Comment (nicktiberi) [inline app/models/collection.rb]: We can use `work.representative_version.published?` here so we don't have to use AASM fields directly. I believe `published?` is an autogenerated method from [AASM](https://github.com/aasm/aasm) (along with `draft?` and `withdrawn?`.

https://github.com/psu-libraries/scholarsphere/blob/main/app/models/work_version.rb#L181-L201

File: app/models/collection.rb (hunk 1 above)
Comment (nicktiberi) [reply app/models/collection.rb]: Maybe `empty?` just for consistency with the `is_empty_bi` Solr field? And also to distinguish it from the AASM draft state for work versions?

File: app/schemas/collection_schema.rb (hunk 2 above)
Comment (nicktiberi) [reply app/schemas/collection_schema.rb:3]: Caveat that I don't know how these schemas are typically set up/organized, but I like `CollectionSchema` here — it seems like a good place to put anything else that may be specific to collections.

**Review by nicktiberi — COMMENTED** [review]
Just a few minor suggestions, but overall this looks great to me!

File: app/schemas/collection_schema.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  def document
+    return {} unless resource.respond_to?(:draft?)
+
+    {
+      is_empty_bi: resource.draft?
```
Comment (nicktiberi) [inline app/schemas/collection_schema.rb]: Just a question for my own Solr understanding — when does `is_empty_bi` get populated for all documents in Solr? Does this just magically happen or do we need to run a full reindex or something to populate this field for all collections so the existing collections with no published works don't appear in search results?

File: app/models/collection.rb (hunk 3 above)
Comment (banukutlu) [reply app/models/collection.rb]: ahh good point, I will use that! thnx!

File: app/schemas/collection_schema.rb (hunk 4 above)
Comment (banukutlu) [reply app/schemas/collection_schema.rb]: right, we need to run full reindex, thanks for bringing it up, I forgot to mention that in my comments.

Comment (banukutlu) [reply app/schemas/collection_schema.rb]: I am open to other suggestions on how to handle it, if you have another idea.

Comment (nicktiberi) [reply app/schemas/collection_schema.rb]: Thanks for confirming! I think this approach makes sense. I can check on QA to see how long it takes to index all the collections. We can do this post launch and some time during off hours if it takes a long time.

Comment (banukutlu) [reply app/schemas/collection_schema.rb]: sounds good 👍🏻

File: app/models/collection.rb (hunk 1 above)
Comment (banukutlu) [reply app/models/collection.rb]: updated in the last commit

**Review by nicktiberi — APPROVED** [review]
LGTM!

**Pull request merged**